x-pack/plugins/alerting/server/mocks.ts

```diff
@@ -64,6 +64,17 @@ const createAlertFactoryMock = {
 
     return mock as unknown as AlertInstanceMock<InstanceState, InstanceContext>;
   },
+  done: <
+    InstanceState extends AlertInstanceState = AlertInstanceState,
+    InstanceContext extends AlertInstanceContext = AlertInstanceContext,
+    ActionGroupIds extends string = string
+  >() => {
+    const mock = {
```

Comment:
**nit:** I see this code is copied from `create` but I put a better way to do this (below) which allows to remove `as unknown as ...`. This allows TypeScript to fail when we add functions to AlertFactoryDoneUtils.

```suggestion
    const mock: jest.Mocked<AlertFactoryDoneUtils<InstanceState, InstanceContext, ActionGroupIds>> = {
```

Reply:
Updated in https://github.com/elastic/kibana/pull/124972/commits/afe0c45743cd5a14e4bff3b5523846d02179058a